app/bot/spam_test.go

```diff
@@ -0,0 +1,122 @@
+package bot
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/radio-t/super-bot/app/bot/mocks"
+)
+
+func TestNewSpamFilter(t *testing.T) {
+	client := &mocks.HTTPClient{}
+	sf := NewSpamFilter("http://localhost", client, false)
+	assert.NotNil(t, sf)
+	assert.Equal(t, "http://localhost", sf.casAPI)
+	assert.Equal(t, client, sf.client)
+	assert.False(t, sf.dry)
+}
+
+func TestSpamFilter_OnMessage(t *testing.T) {
+	tests := []struct {
+		name           string
+		mockResp       string
+		mockStatusCode int
+		dryMode        bool
+		expectedResp   Response
+	}{
+		{
+			name:           "User is not a spammer",
+			mockResp:       `{"ok": true, "description": "Not a spammer"}`,
+			mockStatusCode: 200,
+			dryMode:        false,
+			expectedResp:   Response{},
+		},
+		{
+			name:           "User is a spammer, not in dry mode",
+			mockResp:       `{"ok": false, "description": "Is a spammer"}`,
+			mockStatusCode: 200,
+			dryMode:        false,
+			expectedResp: Response{
+				Text:          "this is spam, go to ban Test User",
```

Comment:
```suggestion
				Text:          "this is spam, go to ban, Test User",
```
Looks confusing without a comma